### PR TITLE
Ubuntu based docker images

### PR DESCRIPTION
**Pull request by mzegla**

### CODE REVIEW

File: src/custom_nodes/east_ocr/utils.hpp [hunk 1]
```diff
@@ -39,7 +39,7 @@ std::vector<T> reorder_to_nhwc(const T* nchwVector, int rows, int cols, int chan
             }
         }
     }
-    return std::move(nhwcVector);
+    return nhwcVector;
```
Comment (dkalinowski) [inline src/custom_nodes/east_ocr/utils.hpp:42]: Do example custom nodes work with OVMS based on ubuntu? Example custom nodes are being built using docker with base image centos

**Review by rasapala — APPROVED** [review]
LGTM

File: src/custom_nodes/east_ocr/utils.hpp (hunk 1 above)
Comment (rasapala) [reply src/custom_nodes/east_ocr/utils.hpp:42]: Custom nodes tests pass.

File: Dockerfile.ubuntu [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+ARG BASE_IMAGE=ubuntu:20.04
+FROM $BASE_IMAGE as base_build
+
+LABEL version="1.0.0"
+
+ARG PROJECT_NAME="OpenVINO Model Server 2021.4"
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:22]: 2022.1?

File: Dockerfile.ubuntu [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+RUN if [ "$ov_use_binary" = "1" ] && [ "$DLDT_PACKAGE_URL" != "" ]; then true ; else exit 0 ; fi ; mkdir -p $TEMP_DIR && cd $TEMP_DIR/ && \
+    wget $DLDT_PACKAGE_URL && \
+    tar -zxf l_openvino_toolkit*.tgz && \
+    cd l_openvino_toolkit*/ && \
+    sed -i 's/decline/accept/g' silent.cfg && \
+    ./install.sh -s silent.cfg --ignore-signature && \
+    rm -Rf $TEMP_DIR $INSTALL_DIR/uninstall* /tmp/* $DL_INSTALL_DIR/documentation $DL_INSTALL_DIR/inference_engine/samples
+
+# apt package
+RUN if [ "$ov_use_binary" = "1" ] && [ "$DLDT_PACKAGE_URL" = "" ] ; then true ; else exit 0 ; fi ; \
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:116]: have both paths been tested?

Comment (dkalinowski) [reply Dockerfile.ubuntu:116]: from binary release and from apt

File: Dockerfile.ubuntu [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+#################### END OF OPENVINO BINARY INSTALL
+
+# Build OpenVINO Model Server
+WORKDIR /ovms
+COPY .bazelrc WORKSPACE /ovms/
+COPY external /ovms/external/
+COPY third_party /ovms/third_party/
+
+ENV BAZEL_LINKLIBS=-l%:libstdc++.a
+RUN apt install -y python-is-python3
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:138]: Why not to move it above? To line 43

File: Dockerfile.ubuntu [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+RUN bash -c "sed -i -e 's|REPLACE_PROJECT_NAME|${PROJECT_NAME}|g' /ovms/src/version.hpp"
+RUN if [ "$ov_use_binary" = "1" ] ; then true ; else exit 0 ; fi ; sed -i -e "s#REPLACE_OPENVINO_NAME#`find /opt/intel/ -maxdepth 1 -type d | grep openvino | cut -d'_' -f2`#g" /ovms/src/version.hpp
+ENV LD_LIBRARY_PATH=${LD_LIBRARY_PATH}:/opt/intel/openvino/deployment_tools/inference_engine/lib/intel64/:/opt/intel/openvino/deployment_tools/ngraph/lib/:/opt/intel/openvino/inference_engine/external/tbb/lib/:/openvino/bin/intel64/Release/lib/:/opt/intel/openvino/opencv/lib/
+
+RUN bazel build ${debug_bazel_flags} --jobs $JOBS //src:ovms
+RUN bazel build ${debug_bazel_flags} --jobs $JOBS //src:libsampleloader.so
+
+RUN cd /ovms/src/test/cpu_extension/ && make
+
+#RUN bazel test ${debug_bazel_flags} --jobs $JOBS --test_summary=detailed --test_output=all //src:ovms_test
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:202]: Why are we skipping unit tests?

File: Dockerfile.ubuntu [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+
+COPY ${ovms_metadata_file} metadata.json
+
+RUN ./bazel-bin/src/./ovms --version
+RUN ./bazel-bin/src/./ovms
```
Comment (dkalinowski) [inline Dockerfile.ubuntu:207]: We should add entrypoint so we dont need to specify one in `docker run`

File: Makefile [hunk 7]
```diff
@@ -53,7 +53,7 @@ OV_SOURCE_BRANCH ?= master
 DLDT_PACKAGE_URL ?=
 OV_USE_BINARY ?= 1
 YUM_OV_PACKAGE ?= intel-openvino-runtime-centos7
-
+APT_OV_PACKAGE ?= intel-openvino-runtime-ubuntu20-2021.3.394
```
Comment (dkalinowski) [inline Makefile:56]: 2021.3?

File: src/custom_nodes/east_ocr/utils.hpp [hunk 8]
```diff
@@ -52,7 +52,7 @@ std::vector<T> reorder_to_nchw(const T* nhwcVector, int rows, int cols, int chan
             }
         }
     }
-    return std::move(nchwVector);
+    return nchwVector;
 }
```
Comment (dkalinowski) [inline src/custom_nodes/east_ocr/utils.hpp:56]: Custom nodes are being built on CentOS7. Do they work with ubuntu OVMS image?

Comment (rasapala) [reply src/custom_nodes/east_ocr/utils.hpp:56]: Yes.

File: Dockerfile.ubuntu (hunk 2 above)
Comment (rasapala) [reply Dockerfile.ubuntu:22]: I think we will change it later on in all dockerfiles.

File: Dockerfile.ubuntu (hunk 3 above)
Comment (rasapala) [reply Dockerfile.ubuntu:116]: Yes.

File: Makefile (hunk 7 above)
Comment (rasapala) [reply Makefile:56]: This is the latest we got. Will update in future.

**Review by rasapala — APPROVED** [review]
LGTM

**Review by dkalinowski — APPROVED** [review]

**Pull request merged**